### PR TITLE
[tcat] TCAT documentation cleanup

### PR DESCRIPTION
**Pull request by canisLupus1313**
Commit adds documentation to tcat CLi commands, also removes incorrect parameter form `otBleSecureTcatStart`.

### CODE REVIEW

**Comment by canisLupus1313** [comment]
@Vyrastas FYI.
Promised docs update.

File: src/cli/cli_tcat.cpp [hunk 1]
```diff
@@ -241,6 +269,20 @@ template <> otError Tcat::Process<Cmd("devid")>(Arg aArgs[])
     return error;
 }
 
+/**
+ * @cli tcat start
+ * @code
+ * tcat start
+ * Done
+ * @endcode
+ * Starts TCAT operation.
```
Comment (Vyrastas) [inline src/cli/cli_tcat.cpp:278]: Need a @par before this line so the command description is pulled into our CLI reference

File: src/cli/cli_tcat.cpp [hunk 2]
```diff
@@ -279,6 +321,15 @@ template <> otError Tcat::Process<Cmd("start")>(Arg aArgs[])
     return error;
 }
 
+/**
+ * @cli tcat stop
+ * @code
+ * tcat stop
+ * Done
+ * @endcode
+ * Stops TCAT operation.
```
Comment (Vyrastas) [inline src/cli/cli_tcat.cpp:330]: Need a @par before this line so the command description is pulled into our CLI reference

**Review by Vyrastas — CHANGES_REQUESTED** [review]
Hi @canisLupus1313 , thanks for doing this!  A few comments

File: src/cli/cli_tcat.cpp [hunk 3]
```diff
@@ -137,6 +137,23 @@ static void HandleBleSecureReceive(otInstance               *aInstance,
...
 }
 
+/**
+ * @cli tcat advid
+ * @code
+ * tcat advid ianapen f378aabb
+ * Done
+ * @endcode
+ * @cparam tcat advid [@ca{id_type}] [@ca{value}]
+ * * The `id_type` has three possible values:
```
[the remark this answers is not in the document]
Comment (canisLupus1313) [reply src/cli/cli_tcat.cpp]: fixed.

File: src/cli/cli_tcat.cpp [hunk 4]
```diff
@@ -137,6 +137,23 @@ static void HandleBleSecureReceive(otInstance               *aInstance,
...
+ *   * `clear` - removes all previously set advertised IDs.
+ *   * `oui24` - sets OUI24 ID type.
+ *   * `oui36` - sets OUI36 ID type.
+ *   * `discriminator` - sets discriminator ID type.
+ *   * `ianapen` - sets ianapen ID type.
```
[the remark this answers is not in the document]
Comment (canisLupus1313) [reply src/cli/cli_tcat.cpp]: fixed.

File: src/cli/cli_tcat.cpp (hunk 1 above)
Comment (canisLupus1313) [reply src/cli/cli_tcat.cpp:278]: I'm not sure if I got the line correctly please double check.

File: src/cli/cli_tcat.cpp (hunk 2 above)
Comment (canisLupus1313) [reply src/cli/cli_tcat.cpp:331]: I'm not sure if I got the line correctly please double check.

File: src/cli/cli_tcat.cpp [hunk 5]
```diff
@@ -241,6 +269,21 @@ template <> otError Tcat::Process<Cmd("devid")>(Arg aArgs[])
...
 }
 
+/**
+ * @cli tcat start
+ * @code
+ * tcat start
+ * Done
+ * @endcode
+ * Starts TCAT operation.
+ * @par
```
Comment (Vyrastas) [inline src/cli/cli_tcat.cpp:279]:
```suggestion
 * @par
 * Starts TCAT operation.
```

File: src/cli/cli_tcat.cpp [hunk 6]
```diff
@@ -279,6 +322,16 @@ template <> otError Tcat::Process<Cmd("start")>(Arg aArgs[])
...
 }
 
+/**
+ * @cli tcat stop
+ * @code
+ * tcat stop
+ * Done
+ * @endcode
+ * Stops TCAT operation.
+ * @par
```
Comment (Vyrastas) [inline src/cli/cli_tcat.cpp:332]:
```suggestion
 * @par
 * Stops TCAT operation.
```

File: src/cli/cli_tcat.cpp (hunk 1 above)
Comment (Vyrastas) [reply src/cli/cli_tcat.cpp:278]: Yeah it should go before the text, not after.  Suggested edits for this and the other.

**Review by Vyrastas — CHANGES_REQUESTED** [review]

File: src/cli/cli_tcat.cpp (hunk 6 above)
Comment (canisLupus1313) [reply src/cli/cli_tcat.cpp]: Fixed.

File: src/cli/cli_tcat.cpp (hunk 5 above)
Comment (canisLupus1313) [reply src/cli/cli_tcat.cpp]: fixed.

**Review by Vyrastas — APPROVED** [review]

**Review by abtink — APPROVED** [review]

File: include/openthread/instance.h [hunk 7]
```diff
@@ -53,7 +53,7 @@ extern "C" {
  * @note This number versions both OpenThread platform and user APIs.
  *
  */
-#define OPENTHREAD_API_VERSION (441)
+#define OPENTHREAD_API_VERSION (442)
```
Comment (jwhui) [inline include/openthread/instance.h:56]: Please rebase and increment :)

**Review by jwhui — APPROVED** [review]
Thanks! 👍🏼

**Pull request merged**